VirtoCommerce.CatalogModule.Data/Services/CatalogSearchServiceImpl.cs

```diff
@@ -133,7 +131,7 @@ protected virtual void SearchCategories(SearchCriteria criteria, SearchResult re
                 //Try to replace sorting columns names
                 TryTransformSortingInfoColumnNames(_categorySortingAliases, sortInfos);
 
-                query = query.OrderBySortInfos(sortInfos);
+                query = query.OrderBySortInfos(sortInfos).ThenBy(q => q.Id);
```

Comment:
According to naming style need to use x for such linq with one argument (in other cases too).